### PR TITLE
bugfixes; add all states

### PR DESCRIPTION
**Pull request by stevebachmeier**
### Description
- *Category*: other
- *JIRA issue*: na
- *Research reference*: na

### Changes and notes
NOTE: THIS IS TO BE MERGED INTO THE `pafs_for_all_states` BRANCH!

These are the changes I'd added to get the the `pafs_for_all_states`
branch to run.

### Verification and Testing
na

### CODE REVIEW

File: src/vivarium_nih_us_cvd/data/loader.py [hunk 1]
```diff
@@ -1045,7 +1045,7 @@ def format_paf_data(entity: str, location: str) -> pd.DataFrame:
         data_keys.POPULATION.STRUCTURE, location
     ).droplevel("location")
 
-    pafs = pd.read_hdf(paths.CALCULATED_PAFS_ROOT + location.lower() +'_pafs.hdf')
```
Comment (stevebachmeier) [inline src/vivarium_nih_us_cvd/data/loader.py:1048]: Bugfix: this link-building didn't work

File: src/vivarium_nih_us_cvd/constants/metadata.py [hunk 2]
```diff
@@ -11,14 +11,63 @@
 CLUSTER_PROJECT = "proj_simscience_prod"
 
 CLUSTER_QUEUE = "all.q"
-MAKE_ARTIFACT_MEM = "10G"
-MAKE_ARTIFACT_CPU = "1"
-MAKE_ARTIFACT_RUNTIME = "3:00:00"
+MAKE_ARTIFACT_MEM = 10  # GB
+MAKE_ARTIFACT_CPU = 1
+MAKE_ARTIFACT_RUNTIME = "6:00:00"
```
Comment (rmudambi) [inline src/vivarium_nih_us_cvd/constants/metadata.py]: Is 6 hours needed?

File: src/vivarium_nih_us_cvd/data/calculated_pafs/alabama_pafs.hdf
Comment (rmudambi) [inline src/vivarium_nih_us_cvd/data/calculated_pafs/alabama_pafs.hdf:1]: As we discussed before, these files shouldn't be in the repository. They should be stored in a directory where we store the actual artifact.

**Review by rmudambi — APPROVED** [review]

File: src/vivarium_nih_us_cvd/constants/metadata.py (hunk 2 above)
Comment (stevebachmeier) [reply src/vivarium_nih_us_cvd/constants/metadata.py]: my data point of 1 for Alabama took 3.6 hours. I'm not sure how much overhead to give it by default. I'll change to 4 for now.

File: src/vivarium_nih_us_cvd/constants/metadata.py [hunk 3]
```diff
@@ -11,14 +11,63 @@
 CLUSTER_PROJECT = "proj_simscience_prod"
 
 CLUSTER_QUEUE = "all.q"
-MAKE_ARTIFACT_MEM = "10G"
-MAKE_ARTIFACT_CPU = "1"
-MAKE_ARTIFACT_RUNTIME = "3:00:00"
+MAKE_ARTIFACT_MEM = 5  # GB
+MAKE_ARTIFACT_CPU = 1
+MAKE_ARTIFACT_RUNTIME = "4:00:00"
```
Comment (mattkappel) [inline src/vivarium_nih_us_cvd/constants/metadata.py:16]: Does it seriously take up to 4 hours to make an artifact?

Comment (stevebachmeier) [reply src/vivarium_nih_us_cvd/constants/metadata.py:16]: Yeah, it took 3.5 for Alabama

Comment (mattkappel) [reply src/vivarium_nih_us_cvd/constants/metadata.py:16]: Wow. All righty then.

**Review by mattkappel — APPROVED** [review]

File: paf_utilities/calculate_pafs.sh [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+locations=(
+# "alabama" \
+# "alaska" \
+# "arizona" \
+# "arkansas" \
+# "california" \
+# "colorado" \
+# "connecticut" \
+# "celaware" \
```
Comment (mattkappel) [inline paf_utilities/calculate_pafs.sh]: the first state is mispelled

File: paf_utilities/calculate_pafs.sh [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+psimulate_cmd="$1"  # run or restart
+
+if [[ -z "$psimulate_cmd" || ("$psimulate_cmd" != "run" && "$psimulate_cmd" != "restart") ]]; then
+    echo "Invalid argument. Please provide either 'run' or 'restart'; provided '$psimulate_cmd'"
+    exit 1
+fi
+
+locations=(
+# "alabama" \
```
Comment (mattkappel) [inline paf_utilities/calculate_pafs.sh]: Should these be commented?

File: paf_utilities/calculate_pafs.sh [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+locations=(
+"alabama" \
+"alaska" \
+"arizona" \
+"arkansas" \
+"california" \
+"colorado" \
+"connecticut" \
+"celaware" \
```
Comment (mattkappel) [inline paf_utilities/calculate_pafs.sh]: The First State is misspelled.

File: src/vivarium_nih_us_cvd/data/loader.py [hunk 7]
```diff
@@ -679,12 +682,12 @@ def is_transition_rate(name: str) -> bool:
...
-#        (source_key == data_keys.LDL_C.PAF)
-#        or (source_key == data_keys.SBP.PAF)
-#        or (source_key == data_keys.BMI.PAF)
-#    )
     is_calculated_paf = False
+    # is_calculated_paf = (
+    #     (source_key == data_keys.LDL_C.PAF)
+    #     or (source_key == data_keys.SBP.PAF)
+    #     or (source_key == data_keys.BMI.PAF)
+    # )
```
Comment (mattkappel) [inline src/vivarium_nih_us_cvd/data/loader.py:690]: You moved this block only, but why is it commented?

**Review by mattkappel — APPROVED** [review]

File: src/vivarium_nih_us_cvd/data/loader.py (hunk 7 above)
Comment (stevebachmeier) [reply src/vivarium_nih_us_cvd/data/loader.py:690]: The current workflow requires this block to be commended/uncommented depending on whether you're generating pafs or not.

File: paf_utilities/calculate_pafs.sh (hunk 6 above)
Comment (stevebachmeier) [reply paf_utilities/calculate_pafs.sh]: oof, good catch thanks.

File: paf_utilities/calculate_pafs.sh [hunk 8]
```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+psimulate_cmd="$1"  # run or restart
+
+if [[ -z "$psimulate_cmd" || ("$psimulate_cmd" != "run" && "$psimulate_cmd" != "restart") ]]; then
+    echo "Invalid argument. Please provide either 'run' or 'restart'; provided '$psimulate_cmd'"
+    exit 1
+fi
+
+locations=(
```
Comment (rmudambi) [inline paf_utilities/calculate_pafs.sh:25]: Nit: can we read this from `constants.metadata.py`? Not worth spending time on it if it's hard.

File: paf_utilities/check_completion.sh [hunk 9]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# This is not a well fleshed out script. Modify as needed!
```
Comment (rmudambi) [inline paf_utilities/check_completion.sh:3]: Should this be called by `calculate_pafs.sh`? Not sure what the workflow is.

File: paf_utilities/paf_runner.sh [hunk 10]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+artifactver="v1-20230613"
+artifactdir="/mnt/team/simulation_science/costeffectiveness/artifacts/vivarium_nih_us_cvd/51-locations"
+
+queue="long.q"
```
Comment (rmudambi) [inline paf_utilities/paf_runner.sh]: Why long.q?

File: paf_utilities/paf_runner.sh [hunk 11]
```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+artifactver="v1-20230613"
+artifactdir="/mnt/team/simulation_science/costeffectiveness/artifacts/vivarium_nih_us_cvd/51-locations"
```
Comment (rmudambi) [inline paf_utilities/paf_runner.sh]: Do these arguments need to be updated in the script itself prior to running? Or can they be passed as arguments to the script?

**Review by rmudambi — APPROVED** [review]

**Review by hussain-jafari — APPROVED** [review]

**Pull request merged**